src/main/java/bio/terra/service/resourcemanagement/azure/AzureResourceDao.java

```diff
@@ -0,0 +1,424 @@
+            return null;
+        }
+
+        String foundApplicationName =
+            storageAccountResource.getApplicationResource().getAzureApplicationDeploymentName();
+        if (!Objects.equal(foundApplicationName, applicationName)) {
+            // there is a storage account with this name in our metadata, but it's for a different app deployment
+            throw new AzureResourceException(
+                String.format("A storage account with this name already exists for a different application: %s, %s",
+                    storageAccountName, applicationName));
+        }
+
+        return storageAccountResource;
+    }
+
+    /**
+     * Fetch an existing storage_account_resource metadata row using the id.
+     * This method expects that there is exactly one row matching the provided resource id.
+     *
+     * @param storageAccountId unique id of a storage account resource
+     * @return a reference to the storage account as a POJO StorageAccountResource
+     * @throws AzureResourceNotFoundException if no storage_account_resource metadata row is found
+     */
+    @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
+    public AzureStorageAccountResource retrieveStorageAccountById(UUID storageAccountId) {
+        MapSqlParameterSource params = new MapSqlParameterSource().addValue("id", storageAccountId);
+        AzureStorageAccountResource storageAccountResource =
+            retrieveStorageAccountBy(sqlStorageAccountRetrievedById, params);
+        if (storageAccountResource == null) {
+            throw new AzureResourceNotFoundException("Storage account not found for id:" + storageAccountId);
+        }
+        return storageAccountResource;
+    }
+
+    /**
+     * Delete the storage_account_resource metadata row associated with the storage account, provided the row is either
+     * unlocked or locked by the provided flight.
+     *
+     * @param storageAccountName name of storage account to delete
+     * @param flightId   flight trying to do the delete
+     * @return true if a row is deleted, false otherwise
+     */
+    @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+    public boolean deleteStorageAccountMetadata(String storageAccountName, String flightId) {
+        String sql = "DELETE FROM storage_account_resource " +
+            "WHERE name = :name AND (flightid = :flightid OR flightid IS NULL)";
+        MapSqlParameterSource params = new MapSqlParameterSource()
+            .addValue("name", storageAccountName)
+            .addValue("flightid", flightId);
+        int numRowsUpdated = jdbcTemplate.update(sql, params);
+        return (numRowsUpdated == 1);
+    }
+
+    private AzureStorageAccountResource retrieveStorageAccountBy(String sql, MapSqlParameterSource params) {
+        List<AzureStorageAccountResource> storageAccountResources =
+            jdbcTemplate.query(sql, params, (rs, rowNum) -> {
```

Comment:
If we only want one, why not use `queryForObject`?

Reply:
I think it's possible because there are cases where it could be more than one?

Reply:
But later down, we throw if there's more than one.

Reply:
Ah wait, I see why we don't use queryForObject. We don't want to throw if nothing is found and queryForObject does that

Reply:
ah gotcha.